server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -255,6 +255,18 @@
         return new HashMap<>(capacity(expectedSize));
     }
 
+    /**
+     * Returns a linked hash map with a capacity sufficient to keep expectedSize elements without being resized.
+     *
+     * @param expectedSize the expected amount of elements in the map
+     * @param <K> the key type
+     * @param <V> the value type
+     * @return a new pre-sized {@link LinkedHashMap}
+     */
+    public static <K, V> Map<K, V> newLinkedHashMapWithExpectedSize(int expectedSize) {
```

Comment:
I would recommend to keep the return type as `LinkedHashMap`. No need to hid implementation as it is already in the name. This will also avoid the type casts in some use cases